### PR TITLE
Renamed ExecutableNetwork to CompiledModel

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Renamed ExecutableNetwork to CompiledModel

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Review by vladimir-dudnik — APPROVED** [review]
good from samples point of view

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#include "pyopenvino/core/containers.hpp"
+#include "pyopenvino/core/infer_request.hpp"
+
+PYBIND11_MAKE_OPAQUE(Containers::TensorIndexMap);
+PYBIND11_MAKE_OPAQUE(Containers::TensorNameMap);
+
+namespace py = pybind11;
+
+void regclass_CompiledModel(py::module m) {
+    py::class_<ov::runtime::CompiledModel, std::shared_ptr<ov::runtime::CompiledModel>> cls(m, "CompiledModel");
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/core/compiled_model.cpp:19]: As new class is introduced it need to be accordingly added to init files in openvino python folder and imported as CompiledModel with "base-overloading". As old class is removed all Python related code will be broken due to imports missing.

Please change this to `CompiledModel`:
 https://github.com/openvinotoolkit/openvino/blob/ec6f57872fb4a28aebec37633813f9fb9ca71683/src/bindings/python/src/openvino/runtime/__init__.py#L65
 
 And adjust ie_api.py `CompiledModel as CompiledModelBase` and any other `ExecutableNetwork` to `CompiledModel` accordingly:
 https://github.com/openvinotoolkit/openvino/blob/ec6f57872fb4a28aebec37633813f9fb9ca71683/src/bindings/python/src/openvino/runtime/ie_api.py#L10
 
 **Probably tests and some other components will also need this rename.**

**Review by jiwaszki — CHANGES_REQUESTED** [review]

File: src/bindings/python/src/pyopenvino/core/compiled_model.cpp (hunk 1 above)
Comment (ilyachur) [reply src/bindings/python/src/pyopenvino/core/compiled_model.cpp:19]: Yes I will fix it. At least CI should show all incorrect classes.

Comment (ilyachur) [reply src/bindings/python/src/pyopenvino/core/compiled_model.cpp:19]: @jiwaszki fixed

File: docs/snippets/ov_common.cpp [hunk 2]
```diff
@@ -71,7 +71,7 @@ int main() {
     //! [ov_api_2_0:get_inputs_outputs]
 
     //! [ov_api_2_0:compile_model]
-    ov::runtime::ExecutableNetwork exec_network = core.compile_model(network, "CPU");
+    ov::runtime::CompiledModel exec_network = core.compile_model(network, "CPU");
```
Comment (ilya-lavrenov) [inline docs/snippets/ov_common.cpp:74]: Suggest to have proper variable name as well in all places

**Review by jiwaszki — COMMENTED** [review]
Looks fine on Python side, please rebase with master to pass CIs and catch possible type errors in python tests.

**Comment by ilyachur** [comment]
> Looks fine on Python side, please rebase with master to pass CIs and catch possible type errors in python tests.
> 
> @ilyachur you can also change naming in this file (if you have time to do so):
> 
> https://github.com/openvinotoolkit/openvino/blob/562d388ad96c4b04a64edab68a6db44917fdf952/src/bindings/python/tests/test_inference_engine/test_executable_network.py#L20

Fixed

File: docs/snippets/ov_common.cpp (hunk 2 above)
Comment (ilyachur) [reply docs/snippets/ov_common.cpp]: Fixed for samples and docs

**Comment by ilyachur** [comment]
@ilya-lavrenov Please take a look

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**